week-3/todo-list/script.js

```diff
@@ -1,25 +1,64 @@
-function populateTodoList(todos) {
+function populateTodoList(todo) {
   let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+  list.textContent = '';
+  for (const item of todo) {
+    const li = document.createElement('li');
+    list.appendChild(li);
+    li.textContent = item.task;
+    //add the check button
+    const span = document.createElement('span');
+    li.appendChild(span);
+        //Add Deadline
+    const date = document.createElement('i');
+    span.appendChild(date);
+    date.textContent=item.deadline;
+    const iCheck = document.createElement('i');
+    span.appendChild(iCheck);
+    iCheck.classList.add('check');
+    iCheck.addEventListener("click", () => {
+    li.classList.toggle('completed')
+    });
+    // add the delete button
+    const iDelete = document.createElement('i');
+    span.appendChild(iDelete);
+    iDelete.classList.add('trash');
+    iDelete.addEventListener("click", () => {
+      li.remove();
+      });
+  }
 }
+const addTodoButton = document.querySelector("#add");
+const input = document.querySelector('#input');
+const date=document.querySelector('#date');
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
+  { task: "Wash the clothes", completed: false, deadline: '2-10-2023' },
+  { task: "Call the doctor", completed: false,  deadline: '5-11-2023'  }
 ];
-
 populateTodoList(todos);
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
+addTodoButton.addEventListener("click", (event) => {
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
-}
+  console.log(date.value);
+  const input = document.querySelector('#input');
+  if(input.value!==''){
+    todos.push({
+      task: input.value,
+      completed: false,
+      deadline:date.value,
+    });
+  }
+  populateTodoList(todos);
```

Comment:
Good job getting the set-state-then-render approach working!

Reply:
Thank you. :)